src/rotatez.jl

```diff
@@ -44,107 +48,95 @@ end
 
 
 """
-    rotate_x!(output, img, tan_θ, workvec, interp)
+    rotate_x!(output, img, tan_θ, interp)
 """
 function rotate_x!(
     output::AbstractMatrix{<:RealU},
     img::AbstractMatrix{<:RealU},
     tan_θ::RealU,
-    workvec::AbstractVector{<:RealU},
     interp::SparseInterpolator,
 )
 
-    len = length(workvec)
+    len = size(img, 1) # size of a square image
     c_y = (len + 1)/2 # center of yi
     idx = 1:len
 
     for i in idx
-        workvec .= idx[i]
-        broadcast!(-, workvec, workvec, c_y)
-        broadcast!(*, workvec, workvec, tan_θ)
-        broadcast!(+, workvec, workvec, idx)
-        linearinterp!(interp, workvec)
+        s = (idx[i] - c_y) * tan_θ + 1
+        e = (idx[i] - c_y) * tan_θ + len
+        linearinterp!(interp, s, e, len)
         mul!((@view output[:, i]), interp, (@view img[:, i])) # need mul! to avoid allocating
     end
     return output
 end
 
 
 """
-    rotate_x_adj!(output, img, tan_θ, workvec, interp)
+    rotate_x_adj!(output, img, tan_θ, interp)
```

Comment:
it looks like `rotate_x_adj!` is identical to `rotate_x!` right?
So we do not need both right?